src/persist-client/src/operators/shard_source.rs

```diff
@@ -179,9 +201,10 @@ where
     V: Debug + Codec,
     D: Semigroup + Codec64 + Send + Sync,
     F: FnMut(&PartStats) -> bool + 'static,
-    G: Scope,
+    G: Scope<Timestamp = R>,
+    R: Refines<T>,
```

Comment:
If there is a reason to keep it this way, can we keep the original bound structure? It saves one generic parameter. i.e have  `G::Timestamp: Refines<T>` and `T: Timestamp + Lattice + Codec64 + TotalOrder`

Reply:
ah thats nice, ill try that!